### ISSUE TITLE
Refinement of the Progressive HashGrid

### ISSUE THREAD

**Issue by yihua7** [threestudio-project/threestudio#323]
Hi, thanks for the excellent work!
I just found that in line [here](https://github.com/threestudio-project/threestudio/blob/607a4be985be817330b8ef79f817da6f255c14e5/threestudio/models/networks.py#L93), changing `enc = enc * self.mask` to `enc = enc * self.mask + enc.detach() * (1 - self.mask)`  is more reasonable and should bring about better quality.
